### PR TITLE
Check for non Junit5 usage in CI

### LINKED ISSUE #2441: Migrate from JUnit 4 to JUnit 5
It would be nice to upgrade the project from the older JUnit 4 to the newer JUnit 5 Jupiter dependency. The migration process would be large so I thought it would be a good idea to have a ticket to track the progress of the migration.

It has been suggested that this task be broken up into various chunks so that pull requests are smaller, more manageable and easier to review.

There are a few general steps that this migration can be broken down into:

> 1. things that can be done while still on JUnit4 to ease the migration
> 2. things to update the parent POM to manage the JUnit dependencies
> 3. (module migration stage 1) things that can be done in each module to move that module from JUnit4 to JUnit5 with the vintage engine
> 4. (module migration stage 2) things that can be done in each module to move that module from JUnit5 vintage to JUnit5 Jupiter engine

_Originally posted by @ctubbsii in https://github.com/apache/accumulo/issues/2427#issuecomment-1022535919_

Step 1 from the list above includes things that overlap between JUnit 4 and 5. An example of this sort of change is #2435. At this point I'm not sure if there are any more tasks that fit into step 1.

Step 2 may only require one PR as well.

The following table (_Originally posted by @ctubbsii in https://github.com/apache/accumulo/issues/2427#issuecomment-1022535919_) may aide in keeping track of stages 3 and 4. This table can be updated to reflect the progress of the migration as it happens, using :heavy_check_mark: or something similar to mark completion.
| Module      | Migrate to JUnit5 vintage | Migrate to JUnit5 jupiter |
| ----------- | ----------- | ----------- |
| assemble |  |  |
| core |  |  |
| hadoop-mapreduce |  |  |
| iterator-test-harness |  |  |
| minicluster |  |  |
| server/base |  |  |
| server/compaction-coordinator |  |  |
| server/compactor |  |  |
| server/gc |  |  |
| server/manager |  |  |
| server/master |  |  |
| server/monitor |  |  |
| server/native |  |  |
| server/tserver |  |  |
| shell |  |  |
| start | |  |
| test |  |  |

### LINKED ISSUE #2586: Add check to ensure new tests are JUnit5 tests
**Is your feature request related to a problem? Please describe.**
Since the recent migration to JUnit5 ([tracked here](https://github.com/apache/accumulo/issues/2441)), there have been several tests that have slipped through the cracks for one reason or another that use JUnit4 imports. For example #2572 and #2585 have been created to fix  JUnit4 tests whose modules have already been converted to JUnit5.

**Describe the solution you'd like**
It would be nice to add a checkstyle rule, or something similar, to warn of JUnit4 usage on certain modules that have already been fully converted to JUnit5. At this point not all modules are fully converted to JUnit5 so we would need to be able to ignore the check on certain modules.

**Describe alternatives you've considered**
Manually searching each module to see if any tests use JUnit4.

I'm not too sure how this can be achieved but to find these tests, I search within a module for the imports that are not `junit.jupiter` which I do in my IDE with the regex "org.junit.[^j]". Something like the import checks done [here](https://github.com/apache/accumulo/blob/main/pom.xml#L1112-L1127) might work for this but I am not sure how to omit specific modules from being checked.

### PR DESCRIPTION
**Pull request by DomGarguilo**
Fixes #2586

This PR adds a script that runs during CI that will fail if any new non-Junit5/Jupiter tests are added.

There are probably some improvements to be made here. I used [find-unapproved-chars.sh](https://github.com/ctubbsii/accumulo/blob/c418fe0ccec75d51a059e4c3bf0c379e09b66fdb/contrib/ci/find-unapproved-chars.sh) as a reference when writing this script.

### CODE REVIEW

File: contrib/ci/find-unapproved-junit4.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+function findalljunit4tests() {
+  # -P for perl matching, -R for recursive, -l for matching files
+  local opts='-PRl'
+  if [[ $1 == 'print' ]]; then
+    # -P for perl matching, -R for recursive, -l for matching files, -H for always showing filenames
+    opts='-PRlH'
+  fi
+  grep "$opts" --include=*.java \
+  --exclude={CompactorTest.java,CompactionCoordinatorTest.java,AccumuloVFSClassLoaderTest.java} \
```
Comment (DomGarguilo) [inline contrib/ci/find-unapproved-junit4.sh:35]: I'm not sure how to refactor this list of excluded files into its own variable. If someone knows how that can be done please let me know.

Comment (ctubbsii) [reply contrib/ci/find-unapproved-junit4.sh]: Fixed in c0bddd233cdd40f3855e7d7b835cffb4b6b71468

**Review by ctubbsii — APPROVED** [review]
@DomGarguilo I made some additional changes. I think it's cool that it caught that one. Please take a look. I think it's good now.

**Review by ctubbsii — APPROVED** [review]
I've tested this a few dozen times every way I can think of. I'm going to go ahead and merge to clear it out of my TODO queue. @DomGarguilo can follow-up with additional comments/improvements as a follow-on, if necessary.

**Comment by DomGarguilo** [comment]
Your changes look good to me. Thanks for the improvements and review, @ctubbsii.

**Pull request merged**